src/velo-fed/BaseFedCCTP.sol

```diff
@@ -31,129 +30,186 @@ interface ICCTP {
-    @param dolaAmount Amount of DOLA to mint
-    @param dolaToSwap Amount of DOLA to swap for USDC
-    @param useCCTP If true, will use CCTP to bridge USDC. If false, will use Base bridge
-    */
-    function expansionAndSwap(uint dolaAmount, uint dolaToSwap, bool useCCTP) external {
+    modifier onlyGov() {
+        if (msg.sender != gov) revert OnlyGov();
+        _;
+    }
+
+    modifier onlyChair() {
         if (msg.sender != chair) revert OnlyChair();
+        _;
+    }
+
+    /**
+     * @notice Mints `dolaAmount` of DOLA, swaps `dolaToSwap` of DOLA to USDC, then transfers all to `aeroFarmer` through base bridge
+     * @param dolaAmount Amount of DOLA to mint
+     * @param dolaToSwap Amount of DOLA to swap for USDC
+     * @param useCCTP If true, will use CCTP to bridge USDC. If false, will use Base bridge
+     * @param swapCallData Data for calling the exchange proxy to swap DOLA for USDC
+     */
+    function expansionAndSwap(
+        uint256 dolaAmount,
+        uint256 dolaToSwap,
+        bool useCCTP,
+        bytes calldata swapCallData
+    ) external onlyChair {
         if (dolaToSwap > dolaAmount) revert SwapMoreDolaThanMinted();
-        
+
         dolaSupply += dolaAmount;
         DOLA.mint(address(this), dolaAmount);
 
-        DOLA.approve(address(curvePool), dolaToSwap);
-        uint usdcAmount = curvePool.exchange_underlying(dolaIndex, usdcIndex, dolaToSwap, dolaToSwap * (PRECISION - maxSlippageBpsDolaToUsdc) / PRECISION / DOLA_USDC_CONVERSION_MULTI);
+        DOLA.approve(exchangeProxy, dolaToSwap);
+        uint256 usdcAmountBefore = USDC.balanceOf(address(this));
 
-        uint dolaToBridge = dolaAmount - dolaToSwap;
+        (bool success, ) = exchangeProxy.call(swapCallData);
+        if (!success) revert SwapFailed();
+
+        uint256 usdcAmountAfter = USDC.balanceOf(address(this));
+        uint256 usdcAmount = usdcAmountAfter - usdcAmountBefore;
+
+        if (
+            usdcAmount <
+            (dolaToSwap * (PRECISION - maxSlippageBpsDolaToUsdc)) /
+                PRECISION /
+                DOLA_USDC_CONVERSION_MULTI
+        ) {
+            revert SlippageTooHigh();
+        }
+
+        uint256 dolaToBridge = dolaAmount - dolaToSwap;
         DOLA.approve(address(baseBridge), dolaToBridge);
-        
-        baseBridge.depositERC20To(address(DOLA), DOLA_BASE, aeroFarmer, dolaToBridge, 200_000, "");
 
-        if(useCCTP){
+        baseBridge.depositERC20To(
+            address(DOLA),
+            DOLA_BASE,
+            aeroFarmer,
+            dolaToBridge,
+            200_000,
+            ""
+        );
+
+        if (useCCTP) {
             USDC.approve(address(CCTP), usdcAmount);
-            CCTP.depositForBurn(usdcAmount, BASE_CCTP_DOMAIN, bytes32(uint256(uint160(aeroFarmer))), address(USDC));
+            CCTP.depositForBurn(
+                usdcAmount,
+                BASE_CCTP_DOMAIN,
+                bytes32(uint256(uint160(aeroFarmer))),
```

Comment:
I absolutely don't understand this design decision on behalf of CCTP. May consider if `abi.encodePacked(aeroFarmer)` is more readable, but I think this is fine in a not fine sort of way.